app/src/main/java/com/duckduckgo/app/di/NetworkModule.kt

```diff
@@ -150,30 +146,16 @@ class NetworkModule {
         return PixelReQueryInterceptor()
     }
 
-    @Provides
-    fun trackerListService(@Named("api") retrofit: Retrofit): TrackerListService =
-        retrofit.create(TrackerListService::class.java)
-
-    @Provides
-    fun httpsUpgradeService(@Named("api") retrofit: Retrofit): HttpsUpgradeService =
-        retrofit.create(HttpsUpgradeService::class.java)
-
+    @SuppressLint("NoRetrofitCreateMethodCallDetector")
```

Comment:
What makes next for the non-caching services... should we also have a `ContributesNonCachingServiceApi` annotation?

Reply:
Could we pass the named parameter? i.e `@ContributesServiceApi(AppScope::class, named="api")` In my head that's better than copy&pasting the codegen for each different type.

Reply:
For `AutoCompleteService`, we want to ensure those results are not cached as they potentially contain sensitive data (like your search queries).

More generally, the idea was to separate the requests into two types:
- non-sensitive requests (like downloading config files) where it's safe to keep this even after clearing all data
- sensitive requests we shouldn't preserve

Originally added here for context. https://github.com/duckduckgo/Android/pull/515

Reply:
This make sense. I will also migrate the non caching then. Neither are irreversible changes plus having both is good for  consistency I guess.